exporters/otlp/otlphttp/envconfig.go

```diff
@@ -15,72 +15,107 @@
 package otlphttp
 
 import (
+	"crypto/tls"
 	"fmt"
+	"io/ioutil"
 	"net/url"
+	"os"
 	"strconv"
 	"strings"
 	"time"
+
+	"go.opentelemetry.io/otel/exporters/otlp/internal/otlpconfig"
 )
 
-func applyEnvConfigs(cfg *config, getEnv func(string) string) *config {
-	opts := getOptionsFromEnv(getEnv)
+func applyEnvConfigs(cfg *config) {
+	e := envOptionsReader{
+		getEnv:   os.Getenv,
+		readFile: ioutil.ReadFile,
+	}
+
+	opts := e.getOptionsFromEnv()
+	for _, opt := range opts {
+		opt.Apply(cfg)
+	}
+}
+
+type envOptionsReader struct {
+	getEnv   func(string) string
+	readFile func(filename string) ([]byte, error)
+}
+
+func (e *envOptionsReader) applyEnvConfigs(cfg *config) {
+	opts := e.getOptionsFromEnv()
 	for _, opt := range opts {
 		opt.Apply(cfg)
 	}
-	return cfg
 }
 
-func getOptionsFromEnv(env func(string) string) []Option {
+func (e *envOptionsReader) getOptionsFromEnv() []Option {
 	var opts []Option
 
 	// Endpoint
-	if v, ok := getEnv(env, "ENDPOINT"); ok {
+	if v, ok := e.getEnvValue("ENDPOINT"); ok {
 		opts = append(opts, WithEndpoint(v))
 	}
-	if v, ok := getEnv(env, "TRACES_ENDPOINT"); ok {
+	if v, ok := e.getEnvValue("TRACES_ENDPOINT"); ok {
 		opts = append(opts, WithTracesEndpoint(v))
 	}
-	if v, ok := getEnv(env, "METRICS_ENDPOINT"); ok {
+	if v, ok := e.getEnvValue("METRICS_ENDPOINT"); ok {
 		opts = append(opts, WithMetricsEndpoint(v))
 	}
 
 	// Certificate File
-	// TODO: add certificate file env config support
+	if path, ok := e.getEnvValue("CERTIFICATE"); ok {
+		if tls, err := e.readTLSConfig(path); err == nil {
```

Comment:
Should we try to report these errors somehow to the user? Maybe the `otel.ErrorHandler`?

Reply:
This is not a blocking question. If we want to do so, I'm happy addressing in another issues/PR.

Reply:
Yes, I did question myself what I should do with these errors. Wasn't aware of `otel.ErrorHandler`, thanks for pointing it out :).

Some examples of the errors that I've added:

```
2021/04/02 17:27:44 failed to configure otlp exporter certificate 'overrode_by_signal_specific': File not found
2021/04/02 17:27:44 failed to configure otlp metrics exporter certificate 'invalid_cert': failed to append certificate to the cert pool
```